teb_local_planner/src/optimal_planner.cpp

```diff
@@ -1239,11 +1239,14 @@ bool TebOptimalPlanner::isTrajectoryFeasible(dwb_critics::ObstacleFootprintCriti
   for (int i=0; i <= look_ahead_idx; ++i)
   {
     teb().Pose(i).toPoseMsg(pose2d);
-    if ( costmap_model->scorePose(pose2d, dwb_critics::getOrientedFootprint(pose2d, footprint_spec)) < 0 ) {
+    if ( costmap_model->scorePose(pose2d, dwb_critics::getOrientedFootprint(pose2d, footprint_spec)) > 0 ) {
```

Comment:
interesting that we need this change, so it seems like the original logic didn't work

Also, further down, `< 0` is still used, could you check? it looks a bit weird

Reply:
costmap_model is of type dwb_critics::ObstacleFootprintCritic, 
/code/ros2_ws/src/navigation2/nav2_dwb_controller/dwb_critics/src/obstacle_footprint.cpp line 87 
in function 
`double ObstacleFootprintCritic::scorePose(const geometry_msgs::msg::Pose2D &, const Footprint & footprint)`
returned valued may never be lower that zero, because it (this function) just collects costmap cell
values where footprint lines lie.
So, from this, yes, it seems like a original logic error.
Futher down the change of the sign also makes sense and running simulation for the slow down before obstacle case works as well.